libs/ui/src/hooks/use-timer-effect.hook.ts

```diff
@@ -0,0 +1,28 @@
+import { EmptyFn, isDefined } from '@rnw-community/shared';
+import { DependencyList, useEffect } from 'react';
+
+export const useTimerEffect = (
+  callback: EmptyFn,
+  refreshInterval: number,
+  deps: DependencyList = [],
+  scrollOffsetY?: number
+) =>
+  useEffect(() => {
+    let interval: NodeJS.Timeout;
+
+    if (isDefined(scrollOffsetY) && scrollOffsetY === 0) {
+      callback();
+
+      interval = setInterval(callback, refreshInterval);
+    } else if (!isDefined(scrollOffsetY)) {
+      callback();
+
+      interval = setInterval(callback, refreshInterval);
```

Comment:
`scrollOffsetY` should  not be related to `useTimerEffect` logic.  
Looks like passing `useTimerEffect` as `deps` will work for you`